app/email.go

```diff
@@ -709,6 +709,31 @@ func (es *EmailService) sendMailWithEmbeddedFiles(to, subject, htmlBody string,
 	return mailservice.SendMailWithEmbeddedFilesUsingConfig(to, subject, htmlBody, embeddedFiles, mailConfig, license != nil && *license.Features.Compliance, "")
 }
 
+func (es *EmailService) InvalidateVerifyEmailTokensForUser(userID string) *model.AppError {
+	tokens, err := es.srv.Store.Token().GetAllTokensByType(TokenTypeVerifyEmail)
+	if err != nil {
+		return model.NewAppError("InvalidateVerifyEmailTokensForUser", "api.user.invalidate_verify_email_tokens.error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	for _, token := range tokens {
+		tokenExtra := struct {
+			UserId string
+			Email  string
+		}{}
+		if err := json.Unmarshal([]byte(token.Extra), &tokenExtra); err != nil {
+			return model.NewAppError("InvalidateVerifyEmailTokensForUser", "api.user.invalidate_verify_email_tokens_parse.error", nil, err.Error(), http.StatusInternalServerError)
```

Comment:
Would it be better to continue to the loop in case of an error and report the error once we finish the loop?

Reply:
Agree with that, the whole thing should fail, but should try to delete as much tokens as possible.